std/range/package.d

```diff
@@ -10195,13 +10195,18 @@ in
             auto result = adds(start, range.length, overflow);
         else static if (isSigned!Enumerator)
         {
-            Largest!(Enumerator, Signed!LengthType) signedLength;
+            alias signed_t = Largest!(Enumerator, Signed!LengthType);
+            signed_t signedLength;
+            //Can length fit in the signed type
+            overflow = range.length > signed_t.max;
+            /*
```

Comment:
The diff is commenting out code?

Reply:
I left it in to show what it's supposed to be replacing, I can get rid if you want. 

I'll do it this evening but I think the original code might have been broken in the first place because of the exception causing overflow to be set to true but then immediately trampling over the value anyway in the checkedint operation. I need to fix it and write a test case that actually overflows.

Reply:
The history of the codebase is preserved in version control, please don't add commented out code.